### ISSUE TITLE
Notification Settings Not Being Honored in OSX

### ISSUE THREAD

**Issue by jlhjlh** [wavebox/waveboxapp#780]
* Wavebox Version: 4
* Operating System & Version: osx 10.12.6
* Account type (if applicable): gmail

**Expected & actual behavior**
Notifications from Gmail show up in the OSX notifications center even though notifications are completely disabled in Wavebox settings. I tested this with the experimental and regular notification provider.

**Is the bug persistent or intermittent?**
Persistent.

![wavebox](https://user-images.githubusercontent.com/10662400/45302759-2b7cae00-b4e2-11e8-8e96-7e59418d9349.png)

**Comment by Thomas101** [comment]
Hi, thanks for reporting. I'm having a little bit of trouble reproducing this, it would be great to have little more info so I can see if I can get to the bottom of it.

* Would you be able to send a screenshot of the notification settings under the gmail account, e.g.
![screen shot 2018-09-11 at 10 46 46](https://user-images.githubusercontent.com/103586/45352271-06567100-b5b0-11e8-960c-bba56316577e.png)

* Would you also be able to take a look under the developer tools to see if any errors have come up in the Gmail window. You can do this by doing the following
    1. Reload your Gmail tab by right clicking on the icon in the left sidebar then `Reload`
    2. Right clicking anywhere in the Gmail window and pick `Inspect`
    3. In the window that appears, click on the `console` tab and send a screenshot. It should look something like this...
![screen shot 2018-09-11 at 10 46 56](https://user-images.githubusercontent.com/103586/45352520-9ac0d380-b5b0-11e8-8a84-9de0dc3133bb.png)

Thanks

**Comment by jcarroll-mediafly** [comment]
for me, I get slack notifications. Even though currently in slack tab talking to the person im getting notifications for.

**Comment by Thomas101** [comment]
@jlhjlh we think we've found the bug. Can you try this to see if it fixes it for you...
* Quit Wavebox
* Remove the `~/Library/Application Support/wavebox/notification_permissions.records`
* Start Wavebox back up again

@jcarroll-mediafly this one's unrelated, but there's a fix in the code already (84958fb5d60cb61068b348ee93753bd646ab6676) and should be out in a new version early next week. Sorry about the duplicates!

**Comment by jacobCRP** [comment]
Hi. Thanks for writing me back.

That file does not exist for me:

![image](https://user-images.githubusercontent.com/18149388/45493150-1e5afb80-b73c-11e8-9726-25f636785d56.png)

**Comment by Thomas101** [comment]
@jacobCRP are you seeing the same issue as **jcarroll-mediafly** or **jlhjlh**?

**Comment by jacobCRP** [comment]
Yes. They're both my accounts.

**Comment by jlhjlh** [comment]
I am jlhjlh. And it happens on my work and home computer.

**Comment by Thomas101** [comment]
Sorry my confusion! Looks like the permission thing that I was looking at before isn't the cause then.

If you head into your Gmail settings via the cog in the top right of the gmail tab....
![screen shot 2018-09-14 at 15 28 50](https://user-images.githubusercontent.com/103586/45556425-26e22d80-b833-11e8-9f26-cc9c5046bc4a.png)
...and then scroll down to `Desktop Notifications`, which setting is selected? Do you see the _Notifications have been disabled in this browser_ message as highlighted?
![screen shot 2018-09-14 at 15 29 03](https://user-images.githubusercontent.com/103586/45556435-2a75b480-b833-11e8-8d23-54c2b6c96780.png)

**Comment by jacobCRP** [comment]
It's enabled.

Disabling now:

**Comment by Thomas101** [comment]
Okay. Did you have the `Notifications have been disabled in this browser` message?

Disabling them isn't a "real" solution as it will disable them if you use Gmail in any other browser. Wavebox should be blocking them by default...

**Comment by jacobCRP** [comment]
Yes, I did receive it. And I don't like notifications in any browser so
that's fine. Still, it's a bug.

**Comment by Thomas101** [comment]
@jcarroll-mediafly the issue with the Slack notifications has just been released in the latest version of Wavebox (4.1.0)

@jacobCRP still looking into the Gmail notifications and will update shortly

**Comment by Thomas101** [comment]
For the original issue, this looks to be isolated to Gmail when a certain configuration is used.

I've added a [KB article](https://wavebox.io/kb/duplicate-gmail-notifications) with some more information and some workarounds until we have a more permanent fix

**Comment by jacobCRP** [comment]
Thank you.

**Comment by texelate** [comment]
I've having this issue on 4.4.0 with MacOS: no notifications. I am using new style gmail, Gmail notifications are switched off in Gmail settings, offline is disabled.

I have notifications enabled in Wavebox and when I press the **Test Notification** button nothing happens with both **Default** and **Enhanced**.

Any ideas how to get them working? They worked fine up until recently. Is it a 4.4.0 issue?

_(I also tried deleting `~/Library/Application Support/wavebox/notification_permissions.records` to no avail)_

**Comment by max-sixty** [comment]
FYI I'm having a related but slightly different issue:

- No offline mode enabled
- GMail notifications disabled from GMail
- Notifications enabled in Wavebox

But I still receive Wavebox notifications for my GMail account.

There's an easy workaround for me - disabling the Wavebox notifications for the GMail account only - but hopefully helpful as a case regardless

**Comment by Thomas101** [comment]
@max-sixty the notification setting in Gmail itself will have no effect as Wavebox pulls the notifications separately so it can continue providing notifications when the account is not running :)

**Comment by max-sixty** [comment]
Thanks as ever @Thomas101

**Comment by atrasatti** [comment]
Sorry if this is OT, but is offline browsing supported in gmail and Gdocs? It looks to me like Wavebox doesn't support the necessary plugin, am I wrong? Happy to file a separate issue if that's the right way

**Comment by Thomas101** [comment]
@atrasatti Gmail offline is, but Google Docs isn't right now (working on an autumn release for that). If you want to track that specifically, feel free to open a new issue - I don't think there's a specific one right now
